src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4700,32 +4700,48 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
             IOperation initializedInstance = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type,
                                                                           operation.ConstantValue, IsImplicit(operation));
 
-            if (operation.Initializer != null)
-            {
-                SpillEvalStack();
-
-                int initializerCaptureId = _availableCaptureId++;
-                AddStatement(new FlowCapture(initializerCaptureId, initializedInstance.Syntax, initializedInstance));
-
-                initializedInstance = GetCaptureReference(initializerCaptureId, initializedInstance);
-                HandleObjectOrCollectionInitializer(operation.Initializer, initializedInstance);
-            }
+            initializedInstance = HandleObjectOrCollectionInitializer(operation.Initializer, initializedInstance);
 
             return initializedInstance;
         }
 
-        private void HandleObjectOrCollectionInitializer(IObjectOrCollectionInitializerOperation initializer, IOperation initializedInstance)
+        public override IOperation VisitTypeParameterObjectCreation(ITypeParameterObjectCreationOperation operation, int? captureIdForResult)
         {
-            IOperation previousInitializedInstance = _currentImplicitInstance;
-            _currentImplicitInstance = initializedInstance;
+            var initializedInstance = new TypeParameterObjectCreationExpression(initializer: null, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
```

Comment:
>operation.IsImplicit [](start = 179, length = 20)

In object creation we use `IsImplicit` (which also checks a flag for forcing implicit). Should it be used here too?

Reply:
Yes, good catch.

---
In reply to: [193148046](https://github.com/dotnet/roslyn/pull/27460#discussion_r193148046) [](ancestors = 193148046)